contracts/stellar-interchain-token-service/src/token_handler.rs

```diff
@@ -43,12 +53,19 @@ pub fn give_token(
     let token_manager = TokenManagerClient::new(env, &token_manager);
 
     match token_manager_type {
+        // For NativeInterchainToken and MintBurnFrom,
+        // use mint_from which allows the token manager to mint tokens on behalf of the token contract
         TokenManagerType::NativeInterchainToken | TokenManagerType::MintBurnFrom => {
             token_manager.mint_from(env, &token_address, recipient, amount)
         }
+
+        // For LockUnlock, transfer previously locked tokens from the token manager to the recipient
         TokenManagerType::LockUnlock => {
             token_manager.transfer(env, &token_address, recipient, amount)
         }
+
+        // For MintBurn, use direct mint where the token manager mints new tokens
+        // This assumes the token manager has minting privileges on the token contract
```

Comment:
```suggestion
        // This assumes the token manager has minting or admin privileges on the token contract, since `mint` interface doesn't indicate who the caller is
```